### PR TITLE
[quant][pt2e] Move batch norm op between eval/train for cuda

### PR DESCRIPTION
**Pull request by andrewor14**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #123957

Summary: Before in `move_exported_model_to_train/eval`, we only
switched the CPU versions of the batch norm op. This commit adds
support for the cuda versions of the op too. Note that this fix
is temporary; we won't have to differentiate between these two
cases once we have batch norm consolidation.

Test Plan:
python test/test_quantization.py -k test_move_exported_model_bn

Reviewers: jerryzh168

Subscribers: jerryzh168, leslie-fang-intel, supriyar

Differential Revision: [D56070054](https://our.internmc.facebook.com/intern/diff/D56070054)

### CODE REVIEW

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D56070054).

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D56070054).

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D56070054).

File: torch/ao/quantization/pt2e/export_utils.py [hunk 1]
```diff
@@ -136,20 +136,29 @@ def bn_eval(
         torch.randn(1),  # bn_running_mean
         torch.randn(1),  # bn_running_var
     )
+
+    # TODO: We shouldn't assume that the model is on cuda just because cuda is
+    # available, but right now there's no other way to differentiate between
+    # the different bn ops. Once batch norm consolidation lands, we won't have
+    # to do this anymore: https://github.com/pytorch/pytorch/pull/119496.
+    is_cuda = torch.cuda.is_available()
```
Comment (jerryzh168) [inline torch/ao/quantization/pt2e/export_utils.py:144]: right, can we get this from the model itself?

Comment (jerryzh168) [reply torch/ao/quantization/pt2e/export_utils.py:144]: e.g. with this: https://github.com/pytorch/pytorch/blob/main/torch/ao/quantization/fx/utils.py#L225

Comment (andrewor14) [reply torch/ao/quantization/pt2e/export_utils.py:144]: oh this is great, I'll update with this util

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D56070054).

**Comment by jerryzh168** [comment]
@andrewor14 do you still want to land this? if so please fix the CI failures

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D56070054).

**Review by jerryzh168 — APPROVED** [review]
looks good, had a comment inline

File: test/quantization/pt2e/test_quantize_pt2e.py [hunk 2]
```diff
@@ -1840,29 +1852,33 @@ def __init__(self):
             def forward(self, x):
                 return self.bn(x)
 
-        example_inputs = (torch.randn(1, 3, 3, 3),)
-        m = M().train()
+        is_cuda = torch.cuda.is_available()
```
[the remark this answers is not in the document]
Comment (jerryzh168) [reply test/quantization/pt2e/test_quantize_pt2e.py:1855]: sorry this is in test, so it's fine

**Comment by andrewor14** [comment]
@andrewor14 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D56070054).
